File: php/examples/ListObjects.php
<?php

use Qcloud\Cos\Client;
use Qcloud\Cos\Exception\ServiceResponseException;

class ListObjects
{
    private $cosClient;

    private $uploadId;
    private $eTag;
    private $versionId;

    // 获取对象列表
    protected function getBucket() {
        $cosClient = $this->cosClient;
        //.cssg-snippet-body-start:[get-bucket]
        try {
            $bucket = "examplebucket-1250000000"; //存储桶名称 格式：BucketName-APPID
            $result = $cosClient->listObjects(array(
                'Bucket' => $bucket
            ));
            // 请求成功
            if (isset($result['Contents'])) {
                foreach ($result['Contents'] as $rt) {
                    print_r($rt);
                }
            }
        } catch (\Exception $e) {
            // 请求失败
            echo($e);
        }
        
        //.cssg-snippet-body-end
    }

    // 获取对象列表
    protected function getBucketComp() {
        $cosClient = $this->cosClient;
        //.cssg-snippet-body-start:[get-bucket-comp]
        try {
            $result = $cosClient->listObjects(array(
                'Bucket' => 'examplebucket-1250000000', //格式：BucketName-APPID
                'Delimiter' => '',
                'EncodingType' => 'url',
                'Marker' => 'doc/picture.jpg',
                'Prefix' => 'doc',
                'MaxKeys' => 1000,
            )); 
            // 请求成功
            print_r($result);
        } catch (\Exception $e) {
            // 请求失败
            echo($e);
        }
        
        //.cssg-snippet-body-end
    }

    // 获取对象列表
    protected function getBucketRecursive() {
        $cosClient = $this->cosClient;
        //.cssg-snippet-body-start:[get-bucket-recursive]
        try {
            $bucket = "examplebucket-1250000000"; //存储桶名称 格式：BucketName-APPID
            $prefix = ''; //列出对象的前缀
            $marker = ''; //上次列出对象的断点
            while (true) {
                $result = $cosClient->listObjects(array(
                    'Bucket' => $bucket,
                    'Marker' => $marker,
                    'MaxKeys' => 1000 //设置单次查询打印的最大数量，最大为1000
                ));
                if (isset($result['Contents'])) {
                    foreach ($result['Contents'] as $rt) {
                        // 打印key
                        echo($rt['Key'] . "\n");
                    }
                }
                $marker = $result['NextMarker']; //设置新的断点
                if (!$result['IsTruncated']) {
                    break; //判断是否已经查询完
                }
            }
        } catch (\Exception $e) {
            echo($e);
        }
        
        //.cssg-snippet-body-end
    }

	//.cssg-methods-pragma

    protected function init() {
        $secretId = "COS_SECRETID"; //"云 API 密钥 SecretId";
        $secretKey = "COS_SECRETKEY"; //"云 API 密钥 SecretKey";
        $region = "COS_REGION"; //设置一个默认的存储桶地域
        $this->cosClient = new Qcloud\Cos\Client(
            array(
                'region' => $region,
                'schema' => 'https', //协议头部，默认为http
                'credentials'=> array(
                    'secretId'  => $secretId ,
                    'secretKey' => $secretKey)));
    }

    public function mListObjects() {
        $this->init();

        // 获取对象列表
        $this->getBucket();

        // 获取对象列表
        $this->getBucketComp();

        // 获取对象列表
        $this->getBucketRecursive();

	    //.cssg-methods-pragma
    }
}
?>

File: Python/examples/list_objects.py
# -*- coding=utf-8

from qcloud_cos import CosConfig
from qcloud_cos import CosS3Client
from qcloud_cos import CosServiceError
from qcloud_cos import CosClientError

secret_id = 'COS_SECRETID'     # 替换为用户的secret_id
secret_key = 'COS_SECRETKEY'     # 替换为用户的secret_key
region = 'COS_REGION'    # 替换为用户的region
token = None               # 使用临时密钥需要传入Token，默认为空,可不填
config = CosConfig(Region=region, SecretId=secret_id, SecretKey=secret_key, Token=token)  # 获取配置对象
client = CosS3Client(config)

# 获取对象列表
def get_bucket():
    #.cssg-snippet-body-start:[get-bucket]
    response = client.list_objects(
        Bucket='examplebucket-1250000000',
        Prefix='folder1'
    )
    
    #.cssg-snippet-body-end

# 获取对象列表
def get_bucket_comp():
    #.cssg-snippet-body-start:[get-bucket-comp]
    response = client.list_objects(
        Bucket='examplebucket-1250000000',
        Prefix='string',
        Delimiter='/',
        Marker='string',
        MaxKeys=100,
        EncodingType='url'
    )
    
    #.cssg-snippet-body-end

# 获取对象列表
def get_bucket_recursive():
    #.cssg-snippet-body-start:[get-bucket-recursive]
    marker = ""
    while True:
        response = client.list_objects(
            Bucket='examplebucket-1250000000',
            Prefix='folder1',
            Marker=marker
        )
        print(response['Contents'])
        if response['IsTruncated'] == 'false':
            break 
        marker = response['NextMarker']
    
    #.cssg-snippet-body-end

#.cssg-methods-pragma


# 获取对象列表
get_bucket()

# 获取对象列表
get_bucket_comp()

# 获取对象列表
get_bucket_recursive()

#.cssg-methods-pragma